src/main/java/nl/tudelft/planningstool/api/v1/UserAssignmentAPI.java

```diff
@@ -34,20 +33,18 @@
      */
     @GET
     public ListResponse<AssignmentResponse> get(@PathParam("userId") String userId) {
-        long now = System.currentTimeMillis();
-
         User user = this.userDAO.getFromUUID(userId);
 
         Collection<Assignment> alreadyPlannedAssignments = user.getOccurrences().stream()
                 .map(UserOccurrence::getAssignment)
                 .collect(Collectors.toSet());
 
-        Collection<Assignment> assignments = user.getCourses().stream()
+        List<Assignment> assignments = user.getCourses().stream()
                 .map((c) -> c.getCourse().getAssignments())
                 .flatMap(Collection::stream)
                 .filter(a -> !alreadyPlannedAssignments.contains(a))
-                .filter(a -> a.getDeadline() > now)
-                .collect(Collectors.toSet());
+                .sorted((one, other) -> Long.compare(one.getDeadline(), other.getDeadline()))
```

Comment:
I personally like to create a static attribute in `Assignment`:
```java
public static final Comparator<Assignment> BY_DEADLINE = ((one, other) -> Long.compare(one.getDeadline(), other.getDeadline();
```

That way you can reuse this, and your lambda gets more readable as well: `.sorted(BY_DEADLINE)` :smile: